components/developer/groovy/files/Makefile

```diff
@@ -0,0 +1,44 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"). You may
+# only use this file in accordance with the terms of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2016 Jim Klimov
+#
+
+#
+# This Makefile installs groovy into location specified by variables below,
+# as defined in common oi-userland framework (this file is included from
+# main recipe Makefile).
+# List of FS objects below coresponds to groovy-2.4.7 layout.
+#
+
+LN_S ?= $(LN) -s
+LN_SYM_REL = $(GNUBIN)/ln -s -r
+
+$(BUILD_DIR_$(BITS))/.installed-groovy: $(BUILD_$(BITS)) $(SOURCE_DIR)/.prep
```

Comment:
I'm still not sure about this additional complexity. You can use it as independent Makefile... 
Set PATH=/usr/gnu/bin:/usr/bin in top level Makefile and you can do something like:
```
SUBDIRS=bin conf embeddable grooid indy lib
GROOVY_VERSION ?= 2.4.7
all: build

build: 

install: 
       mkdir -p $(DESTDIR)/usr/share/grovy-$(GROOVY_VERSION)
       mkdir -p $(DESTDIR)/usr/bin
       cp -rf $(SUBDIRS) $(DESTDIR)/usr/share/grovy-$(GROOVY_VERSION)
       ln -rs $(DESTDIR)/usr/share/grovy-$(GROOVY_VERSION)/bin/* $(DESTDIR)/usr/bin
```
I think this is a bit prettier as Makefile is independent and shouldn't have any knowledge of our build system.